### PR TITLE
chore: add filter status list for api get employee

### PR DESCRIPTION
**Pull request by datphamcode295**
#### What's this PR do?

- [x] Add filter by status list for api get employee

#### What are the relevant Git tickets?
https://www.notion.so/dwarves/Add-filter-by-statuses-list-for-API-get-Employees-fd3f3ca777ba4c43b07aeecd5186effa

#### Screenshots (if appropriate)
image

#### Request
```
curl --location --request GET 'http://localhost:8200/api/v1/employees?preload=false&workingStatuses=contractor&workingStatuses=probation' \
--header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2OTkzMjExNDIsImlkIjoiMjY1NTgzMmUtZjAwOS00YjczLWE1MzUtNjRjM2EyMmU1NThmIiwiYXZhdGFyIjoiaHR0cHM6Ly9zMy1hcC1zb3V0aGVhc3QtMS5hbWF6b25hd3MuY29tL2ZvcnRyZXNzLWltYWdlcy81MTUzNTc0Njk1NjYzOTU1OTQ0LnBuZyIsImVtYWlsIjoidGhhbmhAZC5mb3VuZGF0aW9uIiwicGVybWlzc2lvbnMiOlsiZW1wbG95ZWVzLnJlYWQiXSwidXNlcl9pbmZvIjpudWxsfQ.GENGPEucSUrILN6tHDKxLMtj0M0REVMUPC7-XhDMpGM' \
--data-raw ''
```

#### Response
```
{
    "page": 0,
    "size": 999,
    "sort": "",
    "total": 2,
    "data": [
        {
            "id": "ecea9d15-05ba-4a4e-9787-54210e3b98ce",
            "createdAt": "2022-11-02T09:50:55.320669Z",
            "updatedAt": "2022-11-02T09:50:55.320669Z",
            "fullName": "Nguyễn Hoàng Huy",
            "displayName": "Huy Nguyen",
            "teamEmail": "huy123@d.foundation",
            "personalEmail": "hoanghuy123@gmail.com",
            "avatar": "https://s3-ap-southeast-1.amazonaws.com/fortress-images/2830497479497502617.png",
            "phoneNumber": "0123456789",
            "address": "chung cư Sunview Town, Gò Dưa, Thủ Đức",
            "mbti": "Defender",
            "gender": "Male",
            "horoscope": "Virgo",
            "birthday": "1990-01-02T00:00:00Z",
            "discordID": "",
            "githubID": "",
            "notionID": "",
            "status": "probation",
            "joinedDate": null,
            "leftDate": null,
            "seniority": null,
            "chapter": null,
            "lineManager": null,
            "positions": [],
            "stacks": [],
            "roles": [],
            "projects": []
        },
        {
            "id": "2655832e-f009-4b73-a535-64c3a22e558f",
            "createdAt": "2022-11-02T09:52:34.586566Z",
            "updatedAt": "2022-11-25T00:54:40.833561Z",
            "fullName": "Phạm Đức Thành",
            "displayName": "Thanh Pham",
            "teamEmail": "thanh@d.foundation",
            "personalEmail": "thanhpham123@gmail.com",
            "avatar": "https://s3-ap-southeast-1.amazonaws.com/fortress-images/5153574695663955944.png",
            "phoneNumber": "0123456788",
            "address": "Phan Huy Ich, Tan Binh District, Ho Chi Minh, Vietnam",
            "mbti": "INTJ-A",
            "gender": "Male",
            "horoscope": "Libra",
            "birthday": "1990-01-02T00:00:00Z",
            "discordID": "",
            "githubID": "",
            "notionID": "",
            "status": "contractor",
            "joinedDate": null,
            "leftDate": null,
            "seniority": null,
            "chapter": null,
            "lineManager": null,
            "positions": [],
            "stacks": [],
            "roles": [],
            "projects": []
        }
    ]
}
```

### CODE REVIEW

File: pkg/handler/employee/employee.go [hunk 1]
```diff
@@ -74,13 +75,19 @@ func (h *handler) List(c *gin.Context) {
...
+	}
+
 	employees, total, err := h.store.Employee.All(h.repo.DB(), employee.GetAllInput{
-		WorkingStatus: query.WorkingStatus,
-		Preload:       query.Preload,
-		Keyword:       query.Keyword,
-		PositionID:    query.PositionID,
-		StackID:       query.StackID,
-		ProjectID:     query.ProjectID,
+		WorkingStatuses: workingStatuses,
```
Comment (nguyennh4522) [inline pkg/handler/employee/employee.go:85]:
```suggestion
		WorkingStatuses: query.GetWorkingStatuses(),
```

File: pkg/handler/employee/request.go [hunk 2]
```diff
@@ -9,12 +9,12 @@ import (
 type GetListEmployeeQuery struct {
 	model.Pagination
 
-	WorkingStatus string `json:"workingStatus" form:"workingStatus"`
-	Preload       bool   `json:"preload" form:"preload,default=true"`
-	PositionID    string `json:"positionID" form:"positionID"`
-	StackID       string `json:"stackID" form:"stackID"`
-	ProjectID     string `json:"projectID" form:"projectID"`
-	Keyword       string `json:"keyword" form:"keyword"`
+	WorkingStatuses string `json:"workingStatuses" form:"workingStatuses"`
```
Comment (nguyennh4522) [inline pkg/handler/employee/request.go:12]: you can try binding with array `[]string`

Comment (datphamcode295) [reply pkg/handler/employee/request.go]: updated

File: pkg/handler/employee/employee.go [hunk 3]
```diff
@@ -74,13 +75,15 @@ func (h *handler) List(c *gin.Context) {
 		"params":  query,
 	})
 
+	pp.Println(query)
```
[the remark this answers is not in the document]
Comment (datphamcode295) [reply pkg/handler/employee/employee.go]: updated

**Pull request merged**